src/meta-srv/src/service/admin/meta.rs

```diff
@@ -0,0 +1,212 @@
+// Copyright 2023 Greptime Team
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::collections::HashMap;
+
+use api::v1::meta::{RangeRequest, RangeResponse};
+use catalog::helper::{CATALOG_KEY_PREFIX, SCHEMA_KEY_PREFIX, TABLE_GLOBAL_KEY_PREFIX};
+use snafu::ResultExt;
+use tonic::codegen::http;
+
+use crate::error::Result;
+use crate::service::admin::HttpHandler;
+use crate::service::store::ext::KvStoreExt;
+use crate::service::store::kv::KvStoreRef;
+use crate::{error, util};
+
+pub struct CatalogsHandler {
+    pub kv_store: KvStoreRef,
+}
+
+pub struct SchemasHandler {
+    pub kv_store: KvStoreRef,
+}
+
+pub struct TablesHandler {
+    pub kv_store: KvStoreRef,
+}
+
+pub struct TableHandler {
+    pub kv_store: KvStoreRef,
+}
```

Comment:
4 different but similar structs with same inner value, just to use `HttpHandler`. Seems too tedious. 

Can we simplify that by change `HttpHandler` or use http path as input param?

Reply:
In the current http framework of admin, the url of our route corresponds to a handler.  https://github.com/evenyag/greptimedb/blob/f8508bb6e7ffb5f409e335134b1473b1a2c1019e/src/meta-srv/src/service/admin.rs#L46  A handler needs to implement handle fn. In addition to the current writing method, I can't think of any other way at present. 0.0